dbms/src/Flash/Mpp/MPPTunnel.cpp

```diff
@@ -70,20 +70,7 @@ MPPTunnel::~MPPTunnel()
     });
     try
     {
-        {
-            std::unique_lock lock(*mu);
-            if (status == TunnelStatus::Finished)
-            {
-                LOG_DEBUG(log, "already finished!");
-                return;
-            }
-
-            /// make sure to finish the tunnel after it is connected
-            waitUntilConnectedOrFinished(lock);
-            finishSendQueue();
-        }
-        LOG_FMT_TRACE(log, "waiting consumer finish!");
-        waitForSenderFinish(/*allow_throw=*/false);
+        close("");
```

Comment:
In MPPTask's destruct, it will close all tunnels, so it is safe to make this change.
https://github.com/pingcap/tiflash/blob/b6536f6af56cc061eec1d65ff82ff79a1c6f81f0/dbms/src/Flash/Mpp/MPPTask.cpp#L67-L74

Reply:
What if one of tunnels throw error in tunnel::close() when closeAllTunnels is called? Is it safe?

Reply:
`MPPTunnel::close()` should not throw error, if it throws error, I think maybe the TiFlash server will be terminated.